### DISCUSSION TITLE
iptables policy ipsec seems not to be enabled on first connection

### DISCUSSION THREAD

**Discussion by robertdahlem** [strongswan/strongswan#880] [Q&A]
Hi,

I have a Strongswan server and client, both on Debian 11, server at IP 1.2.3.4.

I seem to be unable to get payload through the first connection after client reboot because `-m policy --pol ipsec --dir out` does not trigger. My last OUTPUT rules are:

`iptables -A OUTPUT -o vmbr0 -m policy --dir out --pol ipsec -m comment --comment VPN-Traffic -j ACCEPT`
`iptables -A OUTPUT -j LOG --log-prefix "!!! OUTPUT-Drop: " --log-tcp-options`

I get a lot of:

`!!! OUTPUT-Drop: IN= OUT=vmbr0 SRC=192.168.189.13 DST=1.2.3.4 LEN=140 TOS=0x00 PREC=0x00 TTL=64 ID=182 DF PROTO=ESP SPI=0xc084460b`

When I add:

`iptables -A OUTPUT -d 1.2.3.4/32 -o vmbr0 -m policy --dir out --pol none -j LOG --log-prefix "!!! Non-VPN-Traffic: " --log-tcp-options`

in front of both rules (regard `--pol none`) I see:

`!!! Non-VPN-Traffic: IN= OUT=vmbr0 SRC=192.168.189.13 DST=1.2.3.4 LEN=140 TOS=0x10 PREC=0x00 TTL=64 ID=33739 DF PROTO=ESP SPI=0xcd931dc2`
`!!! OUTPUT-Drop:   IN= OUT=vmbr0 SRC=192.168.189.13 DST=1.2.3.4 LEN=140 TOS=0x10 PREC=0x00 TTL=64 ID=33739 DF PROTO=ESP SPI=0xcd931dc2`

So it seems these packets are not matching `-m policy --dir out --pol ipsec`. Only when I vici terminate/initiate the SAs the policy rule ACCEPTs the traffic.

How can I diagnose on which traffic the kernel applies the ipsec policy? And what seems to be going wrong here?

Kind regards,
Robert
[swanctl.conf.client.txt](https://github.com/strongswan/strongswan/files/8010618/swanctl.conf.client.txt)
[swanctl.conf.server.txt](https://github.com/strongswan/strongswan/files/8010619/swanctl.conf.server.txt)

**Comment by tobiasbrunner** [comment]
Any errors in the log for the initial connection? For instance, about route installation? (Without a route that forces `172.20.1.5` or an IP from `172.20.105.0/24` as source, there won't be a policy match. The source IP in the other log message is `192.168.189.13`, which won't match.)

**Reply by robertdahlem** [reply]
Log analysis showed that the first connection did not detect "local host is behind NAT". So I've been hit by #708 again and will configure `encap=yes`. Sorry for stirring that up again.

**Reply by tobiasbrunner** [reply]
Please see #861.
